src/utils/ParseRouteUtilsV3.js

```diff
@@ -0,0 +1,680 @@
+
+      // Discard routes with over 1 hours time waiting between each direction
+      if (previousDirection) {
+        const prevEndTime = Date.parse(previousDirection.endTime);
+        if (prevEndTime + ONE_HOUR_IN_MS < startTime) {
+          return null;
+        }
+      }
+    }
+
+    if (direction.type === DIRECTION_TYPE.WALK) {
+      totalDistanceWalking += direction.distance;
+    }
+    previousDirection = direction;
+  }
+
+  // if a bus route has more walking distance than the walking route, discard route
+  // or route has 0 directions
+  if (totalDistanceWalking > maxWalkingDistance || route.directions.length === 0) {
+    return null;
+  }
+
+  return adjustRouteTimesIfNecessary(route);
+}
+
+/**
+ * If the route arrivalTime and departureTime are within the same minute, increase the
+ * arrivalTime by one minute
+ * @param route
+ * @returns {Object}
+ */
+function adjustRouteTimesIfNecessary(route: Object): Object {
+  const departureDate = new Date(route.departureTime);
+  const arrivalDate = new Date(route.arrivalTime);
+
+  const isWithinMinute = departureDate.getTime() + ONE_MIN_IN_MS > arrivalDate.getTime();
+  const isSameMinute = departureDate.getMinutes() === arrivalDate.getMinutes();
+
+  if (isWithinMinute && isSameMinute) {
+    route.arrivalTime = convertMillisecondsToISOString(arrivalDate.getTime() + ONE_MIN_IN_MS);
+  }
+  return route;
+}
+
+/**
+ * Returns coordinate objects with lat/long after taking in Graphhopper point arrays.
+ *
+ * @param startPathPoints
+ * @param endPathPoints
+ * @returns {Object}
+ */
+function getStartEndCoords(startPathPoints: Object, endPathPoints: Object): { startCoords: Object, endCoords: Object } {
+  const startCoords = {
+    lat: startPathPoints.coordinates[0][1],
+    long: startPathPoints.coordinates[0][0],
+  };
+  const endCoords = {
+    lat: endPathPoints.coordinates[endPathPoints.coordinates.length - 1][1],
+    long: endPathPoints.coordinates[endPathPoints.coordinates.length - 1][0],
+  };
+  return { startCoords, endCoords };
+}
+
+/**
+ * Returns boundingBox from a Graphhopper path
+ *
+ * @param startPathPoints
+ * @param endPathPoints
+ * @returns {Object}
+ */
+function generateBoundingBox(path: Object): Object {
+  return {
+    minLat: path.bbox[1],
+    minLong: path.bbox[0],
+    maxLat: path.bbox[3],
+    maxLong: path.bbox[2],
+  };
+}
+
+/**
+ * Returns the distance between [startCoords] and [endCoords] in miles.
+ * NOTE: Taken from https://www.geodatasource.com/developers/javascript
+ * @param startCoords
+ * @param endCoords
+ * @returns {number}
+ */
+function getDistanceInMiles(startCoords: Object, endCoords: Object): number {
+  const startLat = startCoords.lat;
+  const startLong = startCoords.long;
+  const endLat = endCoords.lat;
+  const endLong = endCoords.long;
+  if (startLat === endLat && startLong === endLong) {
+    return 0;
+  }
+
+  // Convert start and end latitude to radians
+  const startLatRadians = Math.PI * startLat / 180;
```

Comment:
isn't this the Haversine formula used in the distanceBetweenPointsMiles helper function